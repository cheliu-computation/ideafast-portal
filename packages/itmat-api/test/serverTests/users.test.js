const request = require('supertest');
const admin = request.agent(global._APP_);
const user = request.agent(global._APP_);
const { connectAdmin, connectUser, disconnectAgent } = require('./loginHelper');
const { WHO_AM_I, ADD_SHORT_CUT } = require('./gql/usersGql');

beforeAll(async () => { //eslint-disable-line no-undef
    await connectAdmin(admin);
    await connectUser(user);
});


describe('USERS API', () => {
    test('Who am I (admin)',  () => admin
        .post('/graphql')
        .send({ query: WHO_AM_I })
        .then(res => {
            expect(res.status).toBe(200);
            expect(res.body.data.whoAmI.username).toBe('admin');
            expect(res.body).toMatchSnapshot();
            return true;
    }));

    test('Who am I (user)', () => user
        .post('/graphql')
        .send({ query: WHO_AM_I })
        .then(res => {
            expect(res.status).toBe(200);
            expect(res.body.data.whoAmI.username).toBe('chon');
            expect(res.body).toMatchSnapshot();
            return true;
    }));

    test('Add shortcut (study + application) (admin)',  () => admin
        .post('/graphql')
        .send({ query: ADD_SHORT_CUT, variables: { study: 'study001', application: 'youtube' } })
        .then(res => {
            expect(res.status).toBe(200);
            expect(res.body.data.addShortCut.username).toBe('admin');
            expect(res.body.data.addShortCut.shortcuts.length).toBe(1);
            expect(res.body.data.addShortCut.shortcuts[0].study).toBe('study001');
            expect(res.body.data.addShortCut.shortcuts[0].application).toBe('youtube');
            expect(res.body).toMatchSnapshot();
            return true;
    }));

    test('Add shortcut (study) (user)', () => user
        .post('/graphql')
        .send({ query: ADD_SHORT_CUT, variables: { study: 'study001' } })
        .then(res => {
            expect(res.status).toBe(200);
            expect(res.body.data.addShortCut.username).toBe('chon');
            expect(res.body.data.addShortCut.shortcuts.length).toBe(1);
            expect(res.body.data.addShortCut.shortcuts[0].study).toBe('study001');
            expect(res.body.data.addShortCut.shortcuts[0].application).toBe(null);
            expect(res.body).toMatchSnapshot();
            return true;
    }));

    test('Add shortcut (study) (admin)',  () => admin
        .post('/graphql')
        .send({ query: ADD_SHORT_CUT, variables: { study: 'study001', application: 'youtube' } })
        .then(res => {
            expect(res.status).toBe(200);
            expect(res.body.data.addShortCut.username).toBe('admin');
            expect(res.body.data.addShortCut.shortcuts.length).toBe(2);
            expect(res.body).toMatchSnapshot();
            return true;
    }));

    test('Add shortcut (study + application) (user)', () => user
        .post('/graphql')
        .send({ query: ADD_SHORT_CUT, variables: { study: 'study001' } })
        .then(res => {
            expect(res.status).toBe(200);
            expect(res.body.data.addShortCut.username).toBe('chon');
            expect(res.body.data.addShortCut.shortcuts.length).toBe(2);
            expect(res.body).toMatchSnapshot();
            return true;
    }));

});