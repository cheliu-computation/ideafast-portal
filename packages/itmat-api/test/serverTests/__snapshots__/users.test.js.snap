// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`USERS API Add shortcut (study + application) (admin) 1`] = `
Object {
  "data": Object {
    "addShortCut": Object {
      "id": "ffdsfioir4jiofpjsidnvkxzjcklas;",
      "shortcuts": Array [
        Object {
          "application": "youtube",
          "id": "0cd0a970-2966-4699-990b-4d732976f8c0",
          "study": "study001",
        },
      ],
      "username": "admin",
    },
  },
}
`;

exports[`USERS API Add shortcut (study + application) (user) 1`] = `
Object {
  "data": Object {
    "addShortCut": Object {
      "id": "11234234231dsfajkdsjhflkdsla",
      "shortcuts": Array [
        Object {
          "application": null,
          "id": "a0679b38-008d-4a1f-8fd3-373f75a8a85d",
          "study": "study001",
        },
        Object {
          "application": null,
          "id": "39d7c656-a3b0-44bf-9061-1055298169af",
          "study": "study001",
        },
      ],
      "username": "chon",
    },
  },
}
`;

exports[`USERS API Add shortcut (study) (admin) 1`] = `
Object {
  "data": Object {
    "addShortCut": Object {
      "id": "ffdsfioir4jiofpjsidnvkxzjcklas;",
      "shortcuts": Array [
        Object {
          "application": "youtube",
          "id": "0cd0a970-2966-4699-990b-4d732976f8c0",
          "study": "study001",
        },
        Object {
          "application": "youtube",
          "id": "b5d7f215-e1a8-4d90-a8b1-dcf0adc57780",
          "study": "study001",
        },
      ],
      "username": "admin",
    },
  },
}
`;

exports[`USERS API Add shortcut (study) (user) 1`] = `
Object {
  "data": Object {
    "addShortCut": Object {
      "id": "11234234231dsfajkdsjhflkdsla",
      "shortcuts": Array [
        Object {
          "application": null,
          "id": "a0679b38-008d-4a1f-8fd3-373f75a8a85d",
          "study": "study001",
        },
      ],
      "username": "chon",
    },
  },
}
`;

exports[`USERS API Who am I (admin) 1`] = `
Object {
  "data": Object {
    "whoAmI": Object {
      "createdBy": "chon",
      "email": "souc@ic.ac.uk",
      "emailNotificationsActivated": false,
      "id": "ffdsfioir4jiofpjsidnvkxzjcklas;",
      "realName": "admin",
      "shortcuts": Array [],
      "type": "ADMIN",
      "username": "admin",
    },
  },
}
`;

exports[`USERS API Who am I (user) 1`] = `
Object {
  "data": Object {
    "whoAmI": Object {
      "createdBy": "admin",
      "email": "souchon@icloud.com",
      "emailNotificationsActivated": false,
      "id": "11234234231dsfajkdsjhflkdsla",
      "realName": "Chon Sou",
      "shortcuts": Array [],
      "type": "STANDARD",
      "username": "chon",
    },
  },
}
`;
